BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/distance/BeaconDistanceCalculator.java

```diff
@@ -23,6 +23,26 @@ public static float calculateDistanceTo(Beacon beacon) {
         return calculateDistanceTo(beacon, beacon.getFilteredRssi());
     }
 
+    /**
+     * Use Pythagoras with altitude to calculate more accurate distance
+     */
+    public static float calculateDistanceWithHeightTo(Beacon beacon, float rssi) {
+        double altitude = beacon.getLocation().getAltitude();
+        // check for altitude
+        if (altitude > 0) {
+            double distance = calculateDistanceTo(beacon, rssi);
+            double delta = Math.pow(distance, 2) - Math.pow(altitude, 2);
+            // check for negative square root
+            if (delta < 0) {
+                return -((float) Math.sqrt(Math.abs(delta)));
```

Comment:
Why should we ever have a negative result? How can a distance be smaller than 0?

Reply:
Delta = distance - altitude and can be negative